libsolidity/formal/CHC.cpp

```diff
@@ -1001,6 +1007,37 @@ set<unsigned> CHC::transactionVerificationTargetsIds(ASTNode const* _txRoot)
 	return verificationTargetsIds;
 }
 
+optional<CHC::CHCNatspecOption> CHC::natspecOptionFromString(string const& _option)
+{
+	static map<string, CHCNatspecOption> options{
+		{"abstract-function-nondet", CHCNatspecOption::AbstractFunctionNondet}
+	};
+	if (options.count(_option))
+		return options.at(_option);
+	return {};
+}
+
+set<CHC::CHCNatspecOption> CHC::smtNatspecTags(FunctionDefinition const& _function)
+{
+	set<CHC::CHCNatspecOption> options;
+	auto const& tags = _function.annotation().docTags;
+	string smtStr = "custom:smtchecker";
+	for (auto it = tags.find(smtStr); it != tags.end() && it->first == smtStr; ++it)
+	{
+		string const& content = it->second.content;
+		if (auto option = natspecOptionFromString(content))
+			options.insert(*option);
+		else
+			m_errorReporter.warning(3130_error, _function.location(), "Unknown option for \"" + smtStr + "\": \"" + content + "\"");
```

Comment:
I wonder if the warning should go in `DocStringAnalyser`.

Reply:
I feel like it's a different layer, and therefore imo it makes more sense to stay here. But I'm fine with either.

Reply:
What I meant by different layer is that Natspec errors are one thing, but this is more like an external tool that uses the custom tags, so I think the warnings should come from different places.